sdk/core/Microsoft.Azure.Core.NewtonsoftJson/src/Serialization/NewtonsoftJsonObjectSerializer.cs

```diff
@@ -0,0 +1,102 @@
+        {
+            _serializer = useDefaultSettings ?
+                JsonSerializer.CreateDefault(settingsOverrides) :
+                JsonSerializer.Create(settingsOverrides);
+        }
+
+        /// <inheritdoc />
+        /// <exception cref="ArgumentNullException"><paramref name="stream"/> or <paramref name="returnType"/> is null.</exception>
+        public override object Deserialize(Stream stream, Type returnType, CancellationToken cancellationToken)
+        {
+            Argument.AssertNotNull(stream, nameof(stream));
+            Argument.AssertNotNull(returnType, nameof(returnType));
+
+            using StreamReader reader = new StreamReader(stream, UTF8NoBOM, true, DefaultBufferSize, true);
+            return _serializer.Deserialize(reader, returnType);
+        }
+
+        /// <inheritdoc />
+        /// <exception cref="ArgumentNullException"><paramref name="stream"/> or <paramref name="returnType"/> is null.</exception>
+        public override ValueTask<object> DeserializeAsync(Stream stream, Type returnType, CancellationToken cancellationToken) =>
+            new ValueTask<object>(Deserialize(stream, returnType, cancellationToken));
+
+        /// <inheritdoc />
+        /// <exception cref="ArgumentNullException"><paramref name="stream"/> or <paramref name="inputType"/> is null.</exception>
+        public override void Serialize(Stream stream, object? value, Type inputType, CancellationToken cancellationToken)
+        {
+            Argument.AssertNotNull(stream, nameof(stream));
+            Argument.AssertNotNull(inputType, nameof(inputType));
+
+            using StreamWriter writer = new StreamWriter(stream, UTF8NoBOM, DefaultBufferSize, true);
+            _serializer.Serialize(writer, value, inputType);
+        }
+
+        /// <inheritdoc />
+        /// <exception cref="ArgumentNullException"><paramref name="stream"/> or <paramref name="inputType"/> is null.</exception>
+        public override ValueTask SerializeAsync(Stream stream, object? value, Type inputType, CancellationToken cancellationToken)
+        {
+            Serialize(stream, value, inputType, cancellationToken);
+            return new ValueTask();
+        }
+
+        /// <inheritdoc/>
+        public string? GetSerializedMemberName(MemberInfo memberInfo)
+        {
+            Argument.AssertNotNull(memberInfo, nameof(memberInfo));
+
+            if (_serializer.ContractResolver.ResolveContract(memberInfo.ReflectedType) is JsonObjectContract contract)
+            {
+                foreach (JsonProperty property in contract.Properties)
+                {
+                    if (!property.Ignored &&
+                        string.Equals(property.UnderlyingName, memberInfo.Name, StringComparison.Ordinal) &&
+                        property.DeclaringType == memberInfo.DeclaringType)
+                    {
+                        return property.PropertyName;
+                    }
+                }
+            }
+
```

Comment:
Might still make sense to cache this with a `Dictionary<MemberInfo, string>` as well.